libs/community/langchain_community/embeddings/optimum_intel.py

```diff
@@ -0,0 +1,175 @@
+from typing import Any, Dict, List, Optional
+
+import pandas as pd
+import torch
+from langchain_core.embeddings import Embeddings
+from langchain_core.pydantic_v1 import BaseModel, Extra
+from tqdm import tqdm
+from transformers import AutoTokenizer
+
+
+class QuantizedBiEncoderEmbeddings(BaseModel, Embeddings):
+    """HuggingFace sentence_transformers embedding models.
+
+        To use, you should have the ``optimum-intel`` python package installed.
+
+        Example:
+            .. code-block:: python
+
+    from langchain_community.embeddings import QuantizedBiEncoderEmbeddings
+
+    model_name = "Intel/bge-small-en-v1.5-rag-int8-static"
+    encode_kwargs = {'normalize_embeddings': True}
+    hf = QuantizedBiEncoderEmbeddings(
+        model_name=model_name,
+        encode_kwargs=encode_kwargs,
+        query_instruction="Represent this sentence for searching relevant passages: "
+    )
+    """
+
+    def __init__(
+        self,
+        max_seq_len: int = 512,
+        pooling_strategy: str = "mean",  # "mean" or "cls"
+        query_instruction: Optional[str] = None,
+        document_instruction: Optional[str] = None,
+        padding: Optional[bool] = True,
+        use_auth_token: Optional[bool] = False,
+        model_kwargs: Optional[Dict] = {},
+        encode_kwargs: Optional[Dict] = {},
+        **kwargs: Any,
+    ):
```

Comment:
can we add a docstring here for all the init params?

Reply:
Thank you for the feedback! I added the parameter docstring descriptions to the __init__ method.

Reply:
@baskaryan are there any other modifications I should do?